From ab68dc90d3a6e3ae660adb65cf8a020d91eb8f09 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Mon, 10 Feb 2014 12:17:18 +0000
Subject: [PATCH 2/4] Check for GNU libc instead of linux for use of execinfo.h

Since execinfo.h is a GNU extension it makes more sense to check for GNU
than to assume that linux is GNU.

This is needed for building on linux with musl libc.

Signed-off-by: Natanael Copa <ncopa@alpinelinux.org>
---
 core/uwsgi.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/uwsgi.c b/core/uwsgi.c
index 67b175b..b3b25ae 100644
--- a/core/uwsgi.c
+++ b/core/uwsgi.c
@@ -1690,7 +1690,7 @@ void uwsgi_plugins_atexit(void) {
 
 void uwsgi_backtrace(int depth) {
 
-#if defined(__linux__) || (defined(__APPLE__) && !defined(NO_EXECINFO)) || defined(UWSGI_HAS_EXECINFO)
+#if defined(__GLIBC__) || (defined(__APPLE__) && !defined(NO_EXECINFO)) || defined(UWSGI_HAS_EXECINFO)
 
 #include <execinfo.h>
 
-- 
1.8.5.3

